controllers/managedocs_controller.go

```diff
@@ -911,6 +916,26 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+// reconcilePrometheusService function wait for prometheus Service
+// to start and sets appropriate annotation for 'service-ca' controller
+func (r *ManagedOCSReconciler) reconcilePrometheusService() error {
+	r.Log.Info("Reconciling Prometheus Service")
+	prometheusService := &corev1.Service{}
+	err := r.Client.Get(r.ctx,
+		types.NamespacedName{Name: prometheusServiceName, Namespace: r.namespace},
+		prometheusService)
```

Comment:
Please use our utils instead of r.client directly.

```suggestion
		prometheusService := &corev1.Service{}
		prometheusService.Name = prometheusServiceName
		prometheusService.Namespace = r.namespace
		if err := r.get(prometheusService); err != nil {
			return fmt.Errorf("Failed to get the prometheus service %s secret %v", prometheusServiceName, r.AddonParamSecretName)
		}
```

Reply:
Why do we need a secondary service if there is one that already exists? Or am I missing something?